experiments/toptagging/experiment.py

```diff
@@ -285,8 +377,15 @@ def _validate(self, step):
         return metrics["bce"]
 
     def _batch_loss(self, batch):
-        y_pred = self.model(batch)
-        loss = self.loss(y_pred, batch.label.to(self.dtype))
+        LOGGER.info(f"The batch is {batch}")
+        if self.cfg.exp_type == "jctagging":
```

Comment:
Your if-statement approach makes it very hard to keep an overview of our different kinds of experiments. Instead of adding if statements to the abstract TaggingExperiment class, you should have this functionality in the JetClassExperiment defined below, specifically in this case you can just overwrite the _batch_loss function of that class.